src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -249,11 +250,9 @@ public boolean perform(AbstractBuild<?, ?> build,
                         //git.push(null);
                     }
                 } catch (FormException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+                    throw new AbortException("Failed to push merge to origin repository.\n" + e.getMessage());
                 } catch (GitException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+                    throw new AbortException("Failed to push merge to origin repository\n" + e.getMessage());
```

Comment:
In all errors that i saw stacktrace was useless. And users run to me because they saw a stracktrace while error message had valid description.